tests/unit/coordinator/plugins/cassandra/test_restore_steps.py

```diff
@@ -42,20 +66,79 @@ async def test_step_start_cassandra(mocker, override_tokens):
+    node_to_backup_index = [0]
+
     def get_result(cl):
-        if cl == base.BackupManifestStep:
-            return backup_manifest
-        if cl == restore_steps.ParsePluginManifestStep:
-            return plugin_manifest
-        raise NotImplementedError(cl)
+        match cl:
+            case base.BackupManifestStep:
+                return backup_manifest
+            case restore_steps.ParsePluginManifestStep:
+                return plugin_manifest
+            case base.MapNodesStep:
+                return node_to_backup_index
+            case _:
+                raise NotImplementedError(cl)
+
+    expected_reqs = [
+        ipc.CassandraStartRequest(
+            tokens=["token1"] if override_tokens else None,
+            replace_address_first_boot="::1" if replace_backup_nodes else None,
+            skip_bootstrap_streaming=True if replace_backup_nodes else None,
+        )
+    ]
 
-    mocker.patch.object(restore_steps, "run_subop")
+    run_subop = mocker.patch.object(restore_steps, "run_subop")
 
-    step = restore_steps.StartCassandraStep(partial_restore_nodes=None, override_tokens=override_tokens)
+    step = restore_steps.StartCassandraStep(
+        replace_backup_nodes=replace_backup_nodes, override_tokens=override_tokens, cassandra_nodes=[_configuration_node(1)]
+    )
     context = SimpleNamespace(get_result=get_result)
-    cluster = SimpleNamespace(nodes=[SimpleNamespace(az="az1")])
+    cluster = SimpleNamespace(nodes=nodes)
     result = await step.run_step(cluster, context)
     assert result is None
+    run_subop.assert_awaited_once_with(
+        cluster,
+        ipc.CassandraSubOp.start_cassandra,
+        nodes=nodes,
+        reqs=expected_reqs,
+        result_class=ipc.NodeResult,
+    )
+
+
+@pytest.mark.asyncio
+async def test_step_stop_replaced_nodes(mocker):
+    # Node 3 is replacing node 1.
+    manifest_nodes = [_manifest_node(1), _manifest_node(2)]
+    cassandra_nodes = [_configuration_node(1), _configuration_node(2), _configuration_node(3)]
+    nodes = [_coordinator_node(1), _coordinator_node(2), _coordinator_node(3)]
+    node_to_backup_index = [None, None, 0]
+    partial_restore_nodes = [_pr_node(backup_index=0, node_index=2)]
+
+    plugin_manifest = CassandraManifest(cassandra_schema=CassandraSchema(keyspaces=[]), nodes=manifest_nodes)
+
+    def get_result(cl):
+        match cl:
+            case restore_steps.ParsePluginManifestStep:
+                return plugin_manifest
+            case base.MapNodesStep:
+                return node_to_backup_index
+            case _:
+                raise NotImplementedError(cl)
+
+    run_subop = mocker.patch.object(restore_steps, "run_subop")
+
+    step = restore_steps.StopReplacedNodesStep(partial_restore_nodes=partial_restore_nodes, cassandra_nodes=cassandra_nodes)
+    context = SimpleNamespace(get_result=get_result)
+    cluster = SimpleNamespace(nodes=nodes)
+    result = await step.run_step(cluster, context)
```

Comment:
There are already similar instances in this file, but why mocking cluster and context ?
The real classes don't have dependencies and can be used in tests without having to
create a custom `get_result`:

```suggestion
    run_subop = mocker.patch.object(restore_steps, "run_subop")
    step = restore_steps.StopReplacedNodesStep(partial_restore_nodes=partial_restore_nodes, cassandra_nodes=cassandra_nodes)
    context = StepsContext()
    context.setResult(restore_steps.ParsePluginManifestStep, plugin_manifest)
    context.setResult(base.MapNodesStep, node_to_backup_index)
    cluster = Cluster(nodes=nodes)
    result = await step.run_step(cluster, context)
```